app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people.update({name: self})
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons_instances = [
+        Person(person["name"],
+               person["age"])
+        for person in people
+    ]
+
+    for person in people:
+        instances_cache = Person.people
+        person_instance = instances_cache[person["name"]]
```

Comment:
Ensure that the `person['name']` key exists in the `instances_cache` dictionary before accessing it. Otherwise, this could raise a `KeyError` if the name is not found.